src/labelformat/mask_utils.py

```diff
@@ -0,0 +1,372 @@
+"""Mask utilities implemented with Pillow and NumPy, without OpenCV."""
+
+from pathlib import Path
+from typing import Generator, List, Tuple, Union
+
+import numpy as np
+from numpy.typing import NDArray
+from PIL import Image, ImageFilter
+
+from labelformat.model.binary_mask_segmentation import BinaryMaskSegmentation
+from labelformat.model.bounding_box import BoundingBox, BoundingBoxFormat
+from labelformat.model.multipolygon import MultiPolygon
+
+# --- I/O and thresholding ---
```

Comment:
Can you add a comment where these functions come from? Something like "Copied from link_to_opencv_file and then vibe-translated to using only numpy." Otherwise I would need to read up on all these functions, what they do and whether they are implemented correctly.

Reply:
Added a comment